File: readdir/readdir.pl
#!/usr/bin/perl -w
opendir(D, "in");
foreach(readdir(D)) {
    print "$_\n";
}

use File::Find;
find({wanted=>sub {print "$_ "}}, "in");

# Find with:
# grep -r -e readdir -e File::Find .

# Fix with:
# foreach(sort(readdir(D)))

# find({wanted=>sub {print "$_ "}, preprocess => sub {sort {$a cmp $b} @_}}, "in");

File: readdir/readdir.rb
#!/usr/bin/ruby
puts Dir.glob("in/*")
puts Dir["in/*"]
puts Dir.children("in/")
puts Dir.entries("in/")
# See also: .foreach .each .each_child .read

# Find with:
# egrep -r -e 'Dir\.(glob|children|entries|foreach|each|read)' -e 'Dir\[' .

# Fix with:
# puts Dir.entries("in/").sort
